### PR TITLE
feat: update `napi` + uncomment a few related tests

### PR DESCRIPTION
**Pull request by jkomyno**
This PR fixes https://github.com/prisma/team-orm/issues/683.
It also contributes to https://github.com/prisma/team-orm/issues/845.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -52,12 +52,12 @@ uuid = { version = "1", features = ["serde"] }
 indoc = "2.0.1"
 itertools = "0.12"
 connection-string = "0.2"
-napi = { version = "2.14.2", default-features = false, features = [
+napi = { version = "2.15.1", default-features = false, features = [
```
Comment (jkomyno) [inline Cargo.toml:55]: This release includes https://github.com/napi-rs/napi-rs/pull/1934.

**Comment by jkomyno** [comment]
> #### ❌ WASM query-engine performance will worsen by 4.66%
> Full benchmark report
> After changes in [db7f5dc](https://github.com/prisma/prisma-engines/commit/db7f5dcf0274e4abef6c2643663559183a6f9c55)

(We are already aware that these benchmarks are a bit funky now - and they are unrelated to this PR)

**Comment by jkomyno** [comment]
Test failures are not relevant to the PR:
- Benchmarks are currently funky (see https://github.com/prisma/prisma-engines/pull/4687#issuecomment-1921583179)
- `QE: integration tests / vitess_8_0 - graphql 2/4` was flaky (see [Slack](https://prisma-company.slack.com/archives/C05DMKC795G/p1706800112946119))
- The CockroachDB failures had been reported by @aqrln already

**Review by aqrln — APPROVED** [review]
✨

**Pull request merged**